### ISSUE TITLE
dynamic change of lambda param in total loss formula

### ISSUE THREAD

**Issue by takumihoriba** [takumihoriba/WatChMaL#2]
It might be better if we can adjust the lambda parameter in the total equation depending on training phase

Current formula `total_loss = main_loss - lammy * adversary_loss`

Considerations
- be able to specify scheduler from yaml file
- but as a starting point, be able to specify (fixed) lammy value

**Comment by takumihoriba** [comment]
`lammy = self.max_lammy * min(1.0, max(0.7, (self.epoch+1) / self.total_epochs))`

Our current need is to try increasing lambda towards the end of training (after some epochs) so that main classifier would focus more on fooling adversary.

`lammy = 1.0` seemed to work. So with this, first 70% of epochs will be trained with lambda of around 1.
